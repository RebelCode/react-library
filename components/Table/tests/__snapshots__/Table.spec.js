import React from "react";
import { render, cleanup } from "@testing-library/react";
import Table from "../../index";

afterEach(cleanup);

describe("Table", () => {
  const Comp = (
    <Table
      cols={[
        {
          id: "name",
          label: "Name",
          render: (feed) => <h1>col</h1>,
        },
      ]}
      rows={[
        {
          id: "name",
          label: "Name",
          render: (feed) => <h1>rows</h1>,
        },
      ]}
    />
  );

  it("should take a snapshot", () => {
    const { asFragment } = render(Comp);

    expect(asFragment(Comp)).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { queryByLabelText } = render(Comp);

    expect(queryByLabelText("Table")).toBeTruthy();
  });
});
