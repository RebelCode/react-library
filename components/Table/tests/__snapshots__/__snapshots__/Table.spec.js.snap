// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should take a snapshot 1`] = `
<DocumentFragment>
  <table
    aria-label="Table"
    class="table"
  >
    <thead
      class="header"
    >
      <tr>
        <th
          class="colHeading alignLeft"
        >
          Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="row"
      >
        <td
          class="cell alignLeft"
        >
          <h1>
            col
          </h1>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
